hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSecondaryIndexPruning.scala

```diff
@@ -192,6 +192,27 @@ class TestSecondaryIndexPruning extends SparkClientFunctionalTestHarness {
         Seq(s"cde${SECONDARY_INDEX_RECORD_KEY_SEPARATOR}row2"),
         Seq(s"def${SECONDARY_INDEX_RECORD_KEY_SEPARATOR}row3")
       )
+      // update the secondary key column after creating multiple secondary indexes
+      spark.sql(s"update $tableName set not_record_key_col = 'xyz' where record_key_col = 'row1'")
+      // validate the secondary index records themselves
+      checkAnswer(s"select key, SecondaryIndexMetadata.isDeleted from hudi_metadata('$basePath') where type=7")(
```

Comment:
I suggested to read the hoodie_partition meta column as well here and ensure we validate record values in diff secondary indexes.

Reply:
I checked the metadata table. since its virtual keys, we don't have any values in there. so we can't even fetch any meta fields. 
we can ignore this feedback then.